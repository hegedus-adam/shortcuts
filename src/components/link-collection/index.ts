export { renderLinkCollection } from './link-collection.component';
