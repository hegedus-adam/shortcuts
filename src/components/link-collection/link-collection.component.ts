import { renderGroup } from '../link-group';

export function renderLinkCollection(data: any) {
  const article = document.createElement('article');

  data.forEach(group => {
    article.appendChild(renderGroup(group));
  });

  return article;
}
