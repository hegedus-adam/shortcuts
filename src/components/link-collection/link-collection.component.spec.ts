import { renderLinkCollection } from './link-collection.component';
import { JSDOM } from 'jsdom';

const mockRenderLinkGroup = jest.fn();
jest.mock('../link-group', () => ({
  renderGroup: (linkGroup: any) => mockRenderLinkGroup(linkGroup),
}));

const dom = new JSDOM();
global.document = dom.window.document;
global.window = dom.window;

describe('renderLinkCollection', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should append link', () => {
    const link = {
      id: 0,
      label: 'Test Label',
      url: 'https://somewhere.to',
    };

    const group = {
      groupLabel: 'Group Name',
      groupId: 0,
      links: [
        link,
      ],
    };

    const data = [
      group,
    ];

    const mockLinkGroup = document.createElement('li');
    mockLinkGroup.innerHTML = 'mock-link'
    mockRenderLinkGroup.mockReturnValue(mockLinkGroup);

    const actual = renderLinkCollection(data);

    expect(actual).toMatchSnapshot();
    expect(mockRenderLinkGroup.mock.calls.length).toEqual(1);
    expect(mockRenderLinkGroup.mock.calls[0]).toEqual([group]);
  });
});
