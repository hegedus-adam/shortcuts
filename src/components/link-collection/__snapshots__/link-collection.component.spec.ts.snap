// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderLinkCollection should append link 1`] = `
<article>
  <li>
    mock-link
  </li>
</article>
`;
